implicits/src-3/upickle/implicits/CaseClassWriter.scala

```diff
@@ -59,11 +64,23 @@ trait CaseClassWriterPiece extends MacrosCommon:
       Writer.merge[T](writers:_*)
   }
 
+  inline def macroEnumW[T](using m: Mirror.Of[T]): Writer[T] = inline m match {
```

Comment:
same comment about folding macroEnumW into macroW, as I mentioned about the reader case